### PR TITLE
add `bevy_reflect` feature + fix `serde` feature use in hex module

### PR DESCRIPTION
**Pull request by uvizhe**
This adds `bevy_reflect` feature as requested in #98.

Also I found old `ser_de` is not changed to `serde` in Hex.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -32,6 +32,11 @@
 //!
 //! - `hexx = { version = "0.7", features = ["packed"] }`
 //!
+//! `hexx` supports [Bevy Reflection](https://docs.rs/bevy_reflect/latest/bevy_reflect) through the
+//! `bevy_reflect` feature. To enable it add the following line to your `Cargo.toml`:
+//!
+//! - `hexx = { version = "0.8", features = ["bevy_reflect"] }`
```
Comment (ManevilleF) [inline src/lib.rs:38]:
```suggestion
//! - `hexx = { version = "0.7", features = ["bevy_reflect"] }`
```

This feature might be released for 0.7.1

**Review by ManevilleF — CHANGES_REQUESTED** [review]
Thank you for your contribution !

One small request, and after that LGTM

**Comment by ManevilleF** [comment]
> Shell I commit to a changelog as well?

The CI is failing because:
- You should add a CHANGELOG entry
- `rustfmt` is unhappy

**Review by alice-i-cecile — APPROVED** [review]
Very useful, and I firmly agree that this should be off by default.

**Comment by ManevilleF** [comment]
> Very useful, and I firmly agree that this should be off by default.

When bevy reflection improves this wil probably be removed entirely and replaced by a bevy_hexx plugin

**Comment by uvizhe** [comment]
Hmm, looks like adding `bevy_reflect` dev dependency solves the issue with mesh_builder example. Should I add it?

**Comment by ManevilleF** [comment]
> Hmm, looks like adding `bevy_reflect` dev dependency solves the issue with mesh_builder example. Should I add it?
> 
> I'm not sure why there's a problem at all. I appreciate it if someone will explain it to me.

That's a weird issue, I cloned your branch and had no issue compiling the examples, I don't understand why this is happening either.
@alice-i-cecile Do you have any idea ?

**Comment by uvizhe** [comment]
@ManevilleF have you tried to remove your Cargo.lock?

**Comment by alice-i-cecile** [comment]
Yeah my guess is caching weirdness leading to inconsistent results.

**Comment by ManevilleF** [comment]
@uvizhe could you try and rebase (or paply the changes of) #101  ? It seems #99 introduced a compilation problem which might be related to that issue with bevy_reflect

File: Cargo.toml [hunk 2]
```diff
@@ -32,6 +34,11 @@ default-features = false
 features = ["derive"]
 optional = true
 
+[dependencies.bevy_reflect]
+version = "0.10"
+default-features = false
```
Comment (ManevilleF) [inline Cargo.toml:39]:
```suggestion
default-features = false
features = ["glam"]
```

**Review by ManevilleF — CHANGES_REQUESTED** [review]
The `glam` feature is missing from `bevy_reflect`

**Review by ManevilleF — APPROVED** [review]

**Comment by ManevilleF** [comment]
I fixed the example and the missing feature. Everything should work now, I wait for the CI and I merge this

**Pull request merged**